eli5/sklearn/transform.py

```diff
@@ -0,0 +1,179 @@
+"""transform_feature_names implementations for scikit-learn transformers
+"""
+
+import itertools
+
+import numpy as np
+from sklearn.pipeline import Pipeline, FeatureUnion
+from sklearn.feature_selection.base import SelectorMixin
+
+from eli5.transform import transform_feature_names
+from eli5.sklearn.utils import get_feature_names as _get_feature_names
+
+
+@transform_feature_names.register(Pipeline)
+def _pipeline_names(est, in_names=None):
+    names = in_names
+    for name, trans in est.steps:
+        if trans is not None:
+            names = transform_feature_names(trans, names)
+    return names
+
+
+@transform_feature_names.register(FeatureUnion)
+def _union_names(est, in_names=None):
+    return ['{}:{}'.format(trans_name, feat_name)
+            for trans_name, trans, _ in est._iter()
+            for feat_name in transform_feature_names(trans, in_names)]
+
+
+@transform_feature_names.register(SelectorMixin)
+def _select_names(est, in_names=None):
+    mask = est.get_support(indices=False)
+    in_names = _get_feature_names(est, feature_names=in_names,
+                                  num_features=len(mask))
+    return [in_names[i] for i in np.flatnonzero(mask)]
+
+
+# XXX Should these generic things be in eli5.transform?
+# I've left them here to make use of eli5.sklearn.utils.get_feature_names
```

Comment:
There are two kinds of "scikit-learningness": things which work with objects from scikit-learn and things which work with objects which follow scikit-learn API ideas. `get_feature_names` (which is used in xgboost as well) is currently in eli5.sklearn because of the second meaning.